service/controller/v3patch1/key/key.go

```diff
@@ -389,9 +391,27 @@ func VPNGatewayName(customObject providerv1alpha1.AzureConfig) string {
 }
 
 func WorkerInstanceName(customObject providerv1alpha1.AzureConfig, instanceID string) string {
-	return fmt.Sprintf("%s-worker-%06s", ClusterID(customObject), instanceID)
+	idB36 := vmssInstanceIDBase36(instanceID)
+	return fmt.Sprintf("%s-worker-%06s", ClusterID(customObject), idB36)
 }
 
 func WorkerVMSSName(customObject providerv1alpha1.AzureConfig) string {
 	return fmt.Sprintf("%s-worker", ClusterID(customObject))
 }
+
+func vmssInstanceIDBase36(instanceID string) string {
+	i, err := strconv.ParseUint(instanceID, 10, 64)
+	if err != nil {
+		// This must be an int according to the documentation linked below.
+		//
+		// We are panicking here to make the API nice. If this is not
+		// an int there is nothing we can really do and we need to
+		// redesign `instance` resource.
+		//
+		//	https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-instance-ids#scale-set-vm-computer-name
+		//
+		panic(fmt.Sprintf("expected VMSS instanceID to be a positive integer number but got %#q", instanceID))
```

Comment:
Just saying. There is a `panic` in this code. I'm not going to address it ever and if this PR is blocked only because of this panic I'll still merge it. I want to resolve issue we have now. Also I don't see any good handling so I don't wan to pollute the code. Feel free to change it if it hurts you.

Reply:
Please create a TODO issue which we track in the board and go.